<!DOCTYPE html>
<html lang="en">
  <head>
    <title>CRISPResso2</title>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="CRISPResso2: Analysis of genome editing outcomes from deep sequencing data">
    <meta name="author" content="Kendell Clement and Luca Pinello">
    <link href='https://fonts.googleapis.com/css?family=Montserrat|Ranga:700' rel='stylesheet' type='text/css'>
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap/5.1.3/css/bootstrap.min.css" integrity="sha512-GQGU0fMMi238uA+a/bdWJfpUGKUkBdgfFdgBm72SUQ6BeyWjoY/ton0tEjH+OSH9iP4Dfh+7HM0I9f5eR0L/4w==" crossorigin="anonymous" referrerpolicy="no-referrer" />    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.3.5/jquery.fancybox.min.css" />
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.0.13/css/all.css" integrity="sha384-DNOHZ68U8hZfKXOrtjWvjxusGo9WQnrNx2sqG0tfsghAvtVlRW3tvkXWZh58N9jp" crossorigin="anonymous">
    <style>
      body {
        font-family: 'Montserrat', sans-serif !important;
      
      }
      @media (max-width:1015px)
      {
        #crispresso_h
        {
          font-size:5rem;
          font-weight:300;
          line-height:1.2;
        }
        #left_help_div
        {
          display:none
        }
        #top_help_div
        {
          display:block
        }
      }
      @media (min-width:1015px)
      {
        #crispresso_h
        {
          font-size:8rem;
          font-weight:300;
          line-height:1.2;
        }
        #left_help_div
        {
          display:block
        }
        #top_help_div
        {
          display:none
        }
      }
    </style>


    <!-- Always force latest IE rendering engine (even in intranet) & Chrome Frame
    Remove this if you use the .htaccess -->
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

    <!-- Replace favicon.ico & apple-touch-icon.png in the root of your domain and delete these references -->
    
    <link rel="shortcut icon" href="http://crispresso.pinellolab.org/static/favicon.ico" />
    

    <!-- Optional JavaScript -->  
    <!-- jQuery first, then Popper.js, then Bootstrap JS -->
    <script src="https://code.jquery.com/jquery-3.3.1.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.3.5/jquery.fancybox.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-ka7Sk0Gln4gmtz2MlQnikT1wXgYsOg+OMhuP+IlRH9sENBO0LRn5q+8nbTov4+1p" crossorigin="anonymous"></script>


  <script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+'://platform.twitter.com/widgets.js';fjs.parentNode.insertBefore(js,fjs);}}(document, 'script', 'twitter-wjs');</script>
  
<style>
.nav-tabs.amp-header {
  border-bottom:none !important;
}

.card-header.amp-header {
  border-bottom:none !important;
}
.nav-tabs.amp-header .nav-link.active {
  background-color:lightsteelblue;
  border-bottom:lightsteelblue;
}

.pre-scrollable {
    max-height: 340px;
    overflow-y: auto;
    background-color: #f8f8f8;
    text-align: left
}

.tab-content.amp-body {
  background-color:lightsteelblue;
}

@media only screen and (max-width: 600px) {
	.jumbotron img {
		width:100%
	}
    .print-only{
      display: none !important;
    }
    .screen-only{
      display: block;
    }
}

@media print {
  .tab-content > .tab-pane {
   display: block !important;
   opacity: 1 !important;
   visibility: visible !important;
   margin-bottom: .5em !important;
 }
 .nav-tabs {
   display:none !important;
   visibility:hidden !important;
 }
 .tab-content.amp-body {
   background-color:transparent !important;
   border:None !important;
 }
 .col-sm-10 {
		margin-left: -15%;
		width: 130% !important;
	}
 .breakinpage {
   clear: both;
   page-break-before: always !important;
   display: block;
 }
 .print-only, .print-only *
 {
       display: block;
   }
 .screen-only, .screen-only *
 {
   display: none !important;
 }
 div {
	border: none !important;
 }
}
</style>





  </head>

  <body class="m-2">
    

  
  
  <div class="container">
      <div class="row pb-2">
        <div class="col-sm-1"></div>
        
          <div class="col-sm-3 crispresso_cup"> <a href='https://crispresso2.pinellolab.org'><img class='img-fluid' src="https://crispresso.pinellolab.partners.org/static/imgs/CRISPResso_justcup.png" width="80%"></a>
        
          </div>
        <div class="col-sm-7" >
          <a href="https://crispresso2.pinellolab.org" style="text-decoration:none !important; color:rgb(33, 37, 41)"><h1 id='crispresso_h' style="font-family:'Ranga', sans-serif">CRISPResso2 </h1></a>
          <h3 style="margin-top: -0.5em;padding-right: 2em;">Analysis of genome editing outcomes from deep sequencing data</h3>
        </div>
        <div class="col-sm-1">
        </div>
      </div>


    

      <div id='top_help_div'>
        <div class="row">
          <div class="col-sm-1"></div>
          <div class='col-sm-10 pb-2 crispresso_body_div'>
           
          </div>
          <div class="col-sm-1"></div>
        </div>
      </div>

      
<div class="col-sm-1"></div>
<center><div class="col-sm-10">
	<div class="jumbotron" style="background:rgba(0,0,0,0.0); padding:0px" >
		<div id='jumbotron_content' >
			<div class='card text-center mb-2'>
				<div class='card-header'>
					
					<h5>nhej_test</h5>
					
					
	
		<div class="alert alert-danger"><strong>Guardrail Warning!</strong> >=1.0% of reads have modifications at the start or end. Total reads: 24344, Irregular reads: 458.</div>
	
		<div class="alert alert-danger"><strong>Guardrail Warning!</strong> >=0.2% of substitutions were outside of the quantification window. Total substitutions: 9093, Substitutions outside window: 1051.</div>
	

					<h5>CRISPResso2 run information</h5>
					<ul class="nav nav-tabs justify-content-center card-header-tabs" id="log-tab" role="tablist">
						<li class="nav-item">
						<button class="nav-link active" id="log_aln-tab" data-bs-toggle="tab" data-bs-target="#log_aln" role="tab" aria-controls="log_aln" aria-selected="true">Alignment statistics</button>
						</li>
						<li class="nav-item">
						<button class="nav-link" id="log_params-tab" data-bs-toggle="tab" data-bs-target="#log_params" role="tab" aria-controls="log_params" aria-selected="false">Run Parameters</button>
						</li>
					</ul>
				</div>
				<div class='card-body'>
					<div class='tab-content'>
						<div class="tab-pane fade show active" id="log_aln" role="tabpanel">
							<div id="fig_plot_1a" class="d-flex flex-column">
    <a href="CRISPResso_on_nhej_test/1a.Read_barplot.pdf"><img src="CRISPResso_on_nhej_test/1a.Read_barplot.png" width='40%'></a>
        
            <label class="labelpadding">Figure 1a: The number of reads in input fastqs, after preprocessing, and after alignment to amplicons.</label>
                <p class="m-0"><small>Data: <a href="CRISPResso_on_nhej_test/CRISPResso_mapping_statistics.txt">Mapping statistics</a></small></p>
</div>
						</div>
						<div class="tab-pane fade text-start screen-only" id="log_params" role="tabpanel">
  <p><strong>CRISPResso version:</strong> 2.3.3</p>
  <p><strong>Run completed:</strong> 2025-08-05 10:00:14</p>
  <p><strong>Amplicon sequence:</strong> <pre class='pre-scrollable'>AATGTCCCCCAATGGGAAGTTCATCTGGCACTGCCCACAGGTGAGGAGGTCATGATCCCCTTCTGGAGCTCCCAACGGGCCGTGGTCTGGTTCATCATCTGTAAGAATGGCTTCAAGAGGCTCGGCTGTGGTT</pre></p>
  
  <p><strong>Command used:</strong> <pre class='pre-scrollable'>/usr/local/bin/CRISPResso -r1 reads_1.fastq.gz -r2 reads_2.fastq.gz -a AATGTCCCCCAATGGGAAGTTCATCTGGCACTGCCCACAGGTGAGGAGGTCATGATCCCCTTCTGGAGCTCCCAACGGGCCGTGGTCTGGTTCATCATCTGTAAGAATGGCTTCAAGAGGCTCGGCTGTGGTT --name nhej_test --output_folder . --min_average_read_quality 0 --min_single_bp_quality 0</pre></p>
  <p><strong>Parameters:</strong> <pre class='pre-scrollable'>allele_plot_pcts_only_for_assigned_reference: False
aln_seed_count: 5
aln_seed_len: 10
aln_seed_min: 2
amplicon_min_alignment_score: 
amplicon_name: Reference
amplicon_seq: AATGTCCCCCAATGGGAAGTTCATCTGGCACTGCCCACAGGTGAGGAGGTCATGATCCCCTTCTGGAGCTCCCAACGGGCCGTGGTCTGGTTCATCATCTGTAAGAATGGCTTCAAGAGGCTCGGCTGTGGTT
annotate_wildtype_allele: 
assign_ambiguous_alignments_to_first_reference: False
auto: False
bam_chr_loc: 
bam_input: 
bam_output: False
base_editor_output: False
bowtie2_index: 
coding_seq: 
config_file: None
conversion_nuc_from: C
conversion_nuc_to: T
crispresso1_mode: False
crispresso_merge: False
debug: False
default_min_aln_score: 60
disable_guardrails: False
discard_guide_positions_overhanging_amplicon_edge: False
discard_indel_reads: False
display_name: 
dsODN: 
dump: False
exclude_bp_from_left: 15
exclude_bp_from_right: 15
expand_allele_plots_by_quantification: False
expand_ambiguous_alignments: False
expected_hdr_amplicon_seq: 
fastp_command: fastp
fastp_options_string: 
fastq_output: False
fastq_r1: reads_1.fastq.gz
fastq_r2: reads_2.fastq.gz
file_prefix: 
flash_command: None
flexiguide_gap_extend_penalty: -2
flexiguide_gap_open_penalty: -20
flexiguide_homology: 80
flexiguide_name: 
flexiguide_seq: None
force_merge_pairs: False
guide_name: 
guide_seq: 
halt_on_plot_fail: False
ignore_deletions: False
ignore_insertions: False
ignore_substitutions: False
keep_intermediate: False
max_paired_end_reads_overlap: None
max_rows_alleles_around_cut_to_plot: 50
min_average_read_quality: 0
min_bp_quality_or_N: 0
min_frequency_alleles_around_cut_to_plot: 0.2
min_paired_end_reads_overlap: 10
min_single_bp_quality: 0
n_processes: 1
name: nhej_test
needleman_wunsch_aln_matrix_loc: EDNAFULL
needleman_wunsch_gap_extend: -2
needleman_wunsch_gap_incentive: 1
needleman_wunsch_gap_open: -20
no_rerun: False
output_folder: .
place_report_in_output_folder: False
plot_histogram_outliers: False
plot_window_size: 20
prime_editing_gap_extend_penalty: 0
prime_editing_gap_open_penalty: -50
prime_editing_nicking_guide_seq: 
prime_editing_override_prime_edited_ref_seq: 
prime_editing_override_sequence_checks: False
prime_editing_pegRNA_extension_quantification_window_size: 5
prime_editing_pegRNA_extension_seq: 
prime_editing_pegRNA_scaffold_min_match_length: 1
prime_editing_pegRNA_scaffold_seq: 
prime_editing_pegRNA_spacer_seq: 
quantification_window_center: -3
quantification_window_coordinates: None
quantification_window_size: 1
samtools_exclude_flags: 0
save_also_png: False
split_interleaved_input: False
stringent_flash_merging: False
suppress_amplicon_name_truncation: False
suppress_plots: False
suppress_report: False
trim_sequences: False
trimmomatic_command: None
trimmomatic_options_string: 
use_legacy_insertion_quantification: False
use_matplotlib: False
verbosity: 3
write_cleaned_report: False
write_detailed_allele_table: False
zip_output: False</pre></p>
  
  <p class='text-center m-0'><small><a href="CRISPResso_on_nhej_test/CRISPResso_RUNNING_LOG.txt">Running log</a></small></p>
</div>
					</div>
				</div>
			</div>

			<div class='card text-center mb-2 breakinpage'>
				<div class='card-header'>
					<h5>Allele assignments</h5>
					<ul class="nav nav-tabs justify-content-center card-header-tabs" id="aln-tab" role="tablist">
						<li class="nav-item">
							<button class="nav-link active" id="aln_pie-tab" data-bs-toggle="tab" data-bs-target="#aln_pie" role="tab" aria-controls="aln_pie" aria-selected="true">Piechart</button>
						</li>
						<li class="nav-item">
							<button class="nav-link" id="aln_bar-tab" data-bs-toggle="tab" data-bs-target="#aln_bar" role="tab" aria-controls="aln_bar" aria-selected="false">Barplot</button>
						</li>
    						
					</ul>
				</div>
				<div class='card-body'>
					<div class="tab-content" id="tabContent">
						<div class="tab-pane fade show active" id="aln_pie" role="tabpanel" aria-labelledby="aln_pie-tab">
							<div id="fig_plot_1b" class="d-flex flex-column">
    <a href="CRISPResso_on_nhej_test/1b.Alignment_pie_chart.pdf"><img src="CRISPResso_on_nhej_test/1b.Alignment_pie_chart.png" width='40%'></a>
        
            <label class="labelpadding">Figure 1b: Alignment and editing frequency of reads as determined by the percentage and number of sequence reads showing unmodified and modified alleles.</label>
                <p class="m-0"><small>Data: <a href="CRISPResso_on_nhej_test/CRISPResso_quantification_of_editing_frequency.txt">Quantification of editing</a></small></p>
</div>
						</div>
						<div class="tab-pane fade" id="aln_bar" role="tabpanel" aria-labelledby="aln_bar-tab">
							<div id="fig_plot_1c" class="d-flex flex-column">
    <a href="CRISPResso_on_nhej_test/1c.Alignment_barplot.pdf"><img src="CRISPResso_on_nhej_test/1c.Alignment_barplot.png" width='40%'></a>
        
            <label class="labelpadding">Figure 1c: Alignment and editing frequency of reads as determined by the percentage and number of sequence reads showing unmodified and modified alleles.</label>
                <p class="m-0"><small>Data: <a href="CRISPResso_on_nhej_test/CRISPResso_quantification_of_editing_frequency.txt">Quantification of editing</a></small></p>
</div>
						</div>
    						
					</div>
				</div>
			</div> 

        		
			

			

			
			

			
			
            		

			
			

			
			

			
			

			
          	<div> 
          	 


				 
				<div>
				

				
					 
					<div>
					

		      				<div class='card text-center mb-2'>
      							<div class='card-header'>
								 
      								<h5>Nucleotide composition</h5>
								
      							</div>
      							<div class='card-body'>

								
									<div class='div_zoom_outer d-none d-md-block' style="height:100px;border:1px solid #DDDDDD;position:relative">
										<div id="zoomview_nucs_Reference" style="position:absolute;left:0;height:100%;width:100%;background-image: url(CRISPResso_on_nhej_test/2a.Nucleotide_percentage_quilt.png);background-size:auto 100%;background-repeat:no-repeat;"></div>
											<div class="d-lg-none" style="overflow-x:scroll;overflow-y:hidden;position:absolute;width:100%;height:100%">
												<img src="CRISPResso_on_nhej_test/2a.Nucleotide_percentage_quilt.png" style='height:100%'>
											</div>
									</div>
									<div style='position:relative;display:inline-block;width:95%'>
										<a href="CRISPResso_on_nhej_test/2a.Nucleotide_percentage_quilt.pdf">
    									<div id='zoomlens_nucs_Reference' class="d-none d-lg-block" style='float: left;position: absolute;left: 0px;top: 0px;z-index: 1000;border: 1px solid #DDDDDD;height:100%;width:10%'></div>
    										<img id='tozoom_nucs_Reference' src="CRISPResso_on_nhej_test/2a.Nucleotide_percentage_quilt.png" width='100%' style='position:relative'></a>
									</div>
									<label class="labelpadding">
										<span class='d-none d-md-block'>Hover your mouse over the bottom image to zoom in on a specific region.</span><br>
											Figure 2a: Nucleotide distribution across amplicon. At each base in the reference amplicon, the percentage of each base as observed in sequencing reads is shown (A = green; C = orange; G = yellow; T = purple). Black bars show the percentage of reads for which that base was deleted. Brown bars between bases show the percentage of reads having an insertion at that position.
									</label>
								
									<p class="m-0"><small>Data: <a href="CRISPResso_on_nhej_test/Nucleotide_frequency_table.txt">Nucleotide frequency table</a></small></p>
								
								
								
  									
								
            		</div>
					</div> 

  						<div class='card text-center mb-2 breakinpage'>
  							<div class='card-header'>
								 
      							<h5>Modification lengths</h5>
								
  								<ul class="nav nav-tabs justify-content-center card-header-tabs" id="aln-tab" role="tablist">
									
  						  			<li class="nav-item">
						  				<button class="nav-link active" data-bs-toggle="tab" data-bs-target="#Reference_3a" role="tab" aria-controls="Reference_3a" aria-selected="true">Summary</button>
  						  			</li>
            								
									
  						  			<li class="nav-item">
						  				<button class="nav-link" data-bs-toggle="tab" data-bs-target="#Reference_3b" role="tab" aria-controls="Reference_3b" aria-selected="false">Indels</button>
  						  			</li>
            						
  								</ul>
  							</div>
  							<div class='card-body'>
								<div class="tab-content" id="pills-tabContent">
									
  						  			<div class="tab-pane fade show active" id="Reference_3a" role="tabpanel">
							  			<div id="fig_plot_3a" class="d-flex flex-column">
    <a href="CRISPResso_on_nhej_test/3a.Indel_size_distribution.pdf"><img src="CRISPResso_on_nhej_test/3a.Indel_size_distribution.png" width='40%'></a>
        <label class="labelpadding">Figure 3a: Frequency distribution of alleles with indels (blue) and without indels (red). Note that histograms are clipped to show 99% of the data. To show all data, run using the parameter '--plot_histogram_outliers'.  (Maximum 85 not shown) (Minimum -53 not shown)</label>
            <p class="m-0"><small>Data: <a href="CRISPResso_on_nhej_test/Indel_histogram.txt">Indel histogram</a></small></p>
</div>
						  			</div>
									

									
  						  			<div class="tab-pane fade" id="Reference_3b" role="tabpanel">
							  			<div id="fig_plot_3b" class="d-flex flex-column">
    <a href="CRISPResso_on_nhej_test/3b.Insertion_deletion_substitutions_size_hist.pdf"><img src="CRISPResso_on_nhej_test/3b.Insertion_deletion_substitutions_size_hist.png" width='100%'></a>
        <label class="labelpadding">Figure 3b: Left panel, frequency distribution of sequence modifications that increase read length with respect to the reference amplicon, classified as insertions (positive indel size). Middle panel, frequency distribution of sequence modifications that reduce read length with respect to the reference amplicon, classified as deletions (negative indel size). Right panel, frequency distribution of sequence modifications that do not alter read length with respect to the reference amplicon, which are classified as substitutions (number of substituted positions shown).</label>
            <p class="m-0"><small>Data: <a href="CRISPResso_on_nhej_test/Insertion_histogram.txt">Insertions frequency</a></small></p>
            <p class="m-0"><small>Data: <a href="CRISPResso_on_nhej_test/Deletion_histogram.txt">Deletions Frequency</a></small></p>
            <p class="m-0"><small>Data: <a href="CRISPResso_on_nhej_test/Substitution_histogram.txt">Substitutions Frequency</a></small></p>
</div>
						  			</div>
									
  								</div>
  							</div> 
  						</div> 

						<div class='card text-center mb-2 breakinpage'>
  							<div class='card-header'>
								 
  								<h5>Indel characterization</h5>
								
  								<ul class="nav nav-tabs justify-content-center card-header-tabs" id="indel-characterization-tabs" role="tablist">
									
  						  			<li class="nav-item">
						  				<button class="nav-link" data-bs-toggle="tab" data-bs-target="#Reference_4a" role="tab" aria-controls="Reference_4a" aria-selected="true">All Modifications Combined</button>
  						  			</li>
            								
									
  						  			<li class="nav-item">
						  				<button class="nav-link active" data-bs-toggle="tab" data-bs-target="#Reference_4b" role="tab" aria-controls="Reference_4b" aria-selected="false">All Modifications by Type</button>
  						  			</li>
            								
									
  						  			<li class="nav-item">
						  				<button class="nav-link" data-bs-toggle="tab" data-bs-target="#Reference_4c" role="tab" aria-controls="Reference_4c" aria-selected="false">Modifications in Quantification Window</button>
  						  			</li>
            								
									
  						  			<li class="nav-item">
						  				<button class="nav-link" data-bs-toggle="tab" data-bs-target="#Reference_4d" role="tab" aria-controls="Reference_4d" aria-selected="false">Indel Lengths</button>
  						  			</li>
            								
									
									
  								</ul>
  							</div> 
  							<div class='card-body'>
  								<div class="tab-content" id="pills-tabContent">
									
									<div class="tab-pane fade" id="Reference_4a" role="tabpanel">
										<div id="fig_plot_4a" class="d-flex flex-column">
    <a href="CRISPResso_on_nhej_test/4a.Combined_insertion_deletion_substitution_locations.pdf"><img src="CRISPResso_on_nhej_test/4a.Combined_insertion_deletion_substitution_locations.png" width='40%'></a>
        <label class="labelpadding">Figure 4a: Combined frequency of any modification across the amplicon. Modifications outside of the quantification window are also shown.</label>
</div>
									</div>
									
									
									<div class="tab-pane fade show active" id="Reference_4b" role="tabpanel">
										<div id="fig_plot_4b" class="d-flex flex-column">
    <a href="CRISPResso_on_nhej_test/4b.Insertion_deletion_substitution_locations.pdf"><img src="CRISPResso_on_nhej_test/4b.Insertion_deletion_substitution_locations.png" width='40%'></a>
        <label class="labelpadding">Figure 4b: Frequency of insertions, deletions, and substitutions across the entire amplicon, including modifications outside of the quantification window.</label>
            <p class="m-0"><small>Data: <a href="CRISPResso_on_nhej_test/Modification_count_vectors.txt">Modification frequency</a></small></p>
</div>
									</div>
									
									
									<div class="tab-pane fade" id="Reference_4c" role="tabpanel">
										<div id="fig_plot_4c" class="d-flex flex-column">
    <a href="CRISPResso_on_nhej_test/4c.Quantification_window_insertion_deletion_substitution_locations.pdf"><img src="CRISPResso_on_nhej_test/4c.Quantification_window_insertion_deletion_substitution_locations.png" width='40%'></a>
        <label class="labelpadding">Figure 4c: Frequency of insertions, deletions, and substitutions across the entire amplicon, considering only modifications that overlap with the quantification window.</label>
            <p class="m-0"><small>Data: <a href="CRISPResso_on_nhej_test/Quantification_window_modification_count_vectors.txt">Modification frequency in quantification window</a></small></p>
</div>
									</div>
									

									
									<div class="tab-pane fade" id="Reference_4d" role="tabpanel">
										<div id="fig_plot_4d" class="d-flex flex-column">
    <a href="CRISPResso_on_nhej_test/4d.Position_dependent_average_indel_size.pdf"><img src="CRISPResso_on_nhej_test/4d.Position_dependent_average_indel_size.png" width='100%'></a>
        <label class="labelpadding">Figure 4d: Position dependent insertion size(left) and deletion size (right), including only modifications that overlap with the quantification window.</label>
</div>
									</div>
									

									

									
  								</div>
  							</div> 
  						</div> 


						

						

						

						

						

						 
					</div>  <!--end amp tab -->
				
				</div>  <!-- end tab content -->
			</div> <!-- end card -->

            <div align="center" class='p-3'>
  
  <button class='btn btn-primary hidden-print' onclick='window.print();'><i class="fas fa-print"></i> Print</button>
</div>
		</div>  <!-- end jumbotron_content -->
	</div>  <!-- end jumbotron -->
</div>  <!-- end column -->
</center>

<div class="col-sm-1"></div>


      
      <div class="row">
      <div class="col-sm-1"></div>
      <footer class="col-sm-10 crispresso_body_div screen-only py-5">
        <p class="text-center">
      If you like CRISPResso2 please support us by citing it in your work: <br>
          Clement K, Rees H, Canver MC, Gehrke JM, Farouni R, Hsu JY, Cole MA, Liu DR, Joung JK, Bauer DE, Pinello L.<br>
          <a href="https://rdcu.be/boxVG" target="_blank">CRISPResso2 provides accurate and rapid genome editing sequence analysis. </a><br>
          Nat Biotechnol. 2019 Mar; 37(3):224-226. doi: 10.1038/s41587-019-0032-3. PubMed PMID: 30809026.
        </p>
        <p class="text-center"><small> &copy; Copyright <a href="http://pinellolab.org">Kendell Clement and Luca Pinello</a></small></p>
        <p class="text-center">
          <small>
        <a href="https://twitter.com/intent/tweet?text=CRISPResso2%20provides%20accurate%20and%20rapid%20genome%20editing%20sequence%20analysis%20%40kendellclement%20%40lucapinello%20%23CRISPR%20%23CRISPResso2"
          class="twitter-share-button" data-via="kendellclement" data-size="large" data-count="none" data-hashtags="CRISPR" data-url="http://crispresso2.pinellolab.org">Tweet about CRISPresso2!</a>
        </small>
        </p>

      </footer>
      <div class="col-sm-1"></div>
    </div>
    
    </div>
    
<script>

function updateZoom(e) {
  /*prevent any other actions that may occur when moving over the image:*/
//  e.preventDefault();
  var img = e.target.imgObj
  var view = e.target.viewObj
  var lens = e.target.lensObj

  if (!lens.hasWidthSet)
  {
    view_height = $('#'+view.id).height()
    view_width = $('#'+view.id).width()
    img_height = $('#'+img.id).height()
    img_width = $('#'+img.id).width()
    lens_height = $('#'+lens.id).height()
    lens_width = $('#'+lens.id).width()

    new_width = img_height * view_width/view_height  //make up for loss of precision
    $('#'+lens.id).outerWidth(new_width)
    lens.hasWidthSet = true

    cx = view_width / new_width

    view.view_height = view_height
    view.view_width = view_width
    img.img_height = img_height
    img.img_width = img_width
    lens.lens_height = lens_height
    lens.lens_width = new_width

  }

  var pos, x, y;
  /*get the cursor's x and y positions:*/
  pos = getCursorPos(e,img);
  /*calculate the position of the lens:*/
  x = pos.x - (lens.lens_width / 2);
  /*prevent the lens from being positioned outside the image:*/
  if (x > img.img_width - lens.lens_width) {x = img.img_width - lens.lens_width;;
  }
  if (x < 0) {x = 0;}
  lens.style.left = x + "px";
  view.style.backgroundPosition = "-" + (x * cx) + "px 0px";
}

function getCursorPos(e,img) {
  var a, x = 0, y = 0;
  e = e || window.event;
  /*get the x and y positions of the image:*/
  a = img.getBoundingClientRect();
  /*calculate the cursor's x and y coordinates, relative to the image:*/
  x = e.pageX - a.left;
  y = e.pageY - a.top;
  /*consider any page scrolling:*/
  x = x - window.pageXOffset;
  y = y - window.pageYOffset;
  return {x : x, y : y};
}

var passiveSupported = false;
try {
  var options = {
    get passive() { // This function will be called when the browser
                    //   attempts to access the passive property.
      passiveSupported = true;
    }
  };

  window.addEventListener("test", options, options);
  window.removeEventListener("test", options, options);
} catch(err) {
  passiveSupported = false;
}

	
		
		view = document.getElementById('zoomview_nucs_Reference');
		img = document.getElementById('tozoom_nucs_Reference');
		lens = document.getElementById('zoomlens_nucs_Reference')

		img.viewObj = view
		img.lensObj = lens
		img.imgObj = img

		lens.viewObj = view
		lens.lensObj = lens
		lens.imgObj = img

		lens.addEventListener("mousemove", updateZoom, passiveSupported? { passive: true } : false);
		img.addEventListener("mousemove", updateZoom, passiveSupported? { passive: true } : false);
		/*and also for touch screens:*/
		lens.addEventListener("touchmove", updateZoom, passiveSupported? { passive: true } : false);
		img.addEventListener("touchmove", updateZoom, passiveSupported? { passive: true } : false);

		
	

</script>





  </body>

</html>